### PR TITLE
Adding @types for "feathers-rethinkdb"

### PR DESCRIPTION
**Pull request by davidfarinha**
Please fill in this template.

- [ X ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ X ] Test the change in your own code. (Compile and run.)
- [ X ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ X ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ X ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ X ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [ X ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ X ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ X ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ X ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by davidfarinha** [comment]
Getting errors in the test "Error: In /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/feathers-rethinkdb/package.json: Don't use a 'package.json' for @types dependencies." since these typings depend on other external typings ("rethinkdbdash") not stored in DefinitelyTyped, a package.json is required for this module and "rethinkdbdash" needs to be added to the types-publisher dependenciesWhitelist.

[A "types-publisher" pull request](https://github.com/Microsoft/types-publisher/pull/476) has been made. Waiting to this complete first before these tests will pass.

File: types/feathers-rethinkdb/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "@types/rethinkdbdash": "github:types/rethinkdbdash"
```
Comment (andy-ms) [inline types/feathers-rethinkdb/package.json]: I don't think we want to add dependencies on github repositories. Those types should be ported here (with the author's permission) or created fresh.

File: types/feathers-rethinkdb/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": false,
```
Comment (andy-ms) [inline types/feathers-rethinkdb/tsconfig.json]: Should be `true` for all new packages

File: types/feathers-rethinkdb/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        "strictNullChecks": true,
+        "strictFunctionTypes": false,
```
Comment (andy-ms) [inline types/feathers-rethinkdb/tsconfig.json]: Should be `true` for all new packages

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/feathers-rethinkdb/package.json (hunk 1 above)
Comment (davidfarinha) [reply types/feathers-rethinkdb/package.json]: OK, in that case I will close this in favor of using [types/rethinkdbdash](https://github.com/types/rethinkdbdash) and [types/rethinkdb](https://github.com/types/rethinkdb) stored outside of DefinitelyTyped

**Pull request closed without merging**